upbc/protoc-gen-upb.cc

```diff
@@ -95,6 +95,17 @@ std::vector<const protobuf::EnumDescriptor*> SortedEnums(
   return enums;
 }
 
+std::vector<int32_t> SortedUniqueEnumNumbers(
+    const protobuf::EnumDescriptor* e) {
+  std::vector<int32_t> values;
+  for (int i = 0; i < e->value_count(); i++) {
+    values.push_back(e->value(i)->number());
+  }
+  std::sort(values.begin(), values.end());
+  std::unique(values.begin(), values.end());
```

Comment:
don't you need to truncate after the unique call?

Reply:
Good catch. I was redundantly achieving the same by using a `flat_hash_set` in the surrounding function. I've added an `erase()` here and removed the redundant code in the caller.